tests/test_main.rs

```diff
@@ -19,9 +19,60 @@ fn test_run_script() {
     Command::cargo_bin("pn")
         .unwrap()
         .current_dir(&temp_dir)
-        .arg("run")
-        .arg("test")
+        .args(["run", "test"])
         .assert()
         .success()
         .stdout("hello world\n");
 }
+
+#[test]
+fn test_workspace_root() {
+    let temp_dir = tempfile::tempdir().unwrap();
+    let temp_dir_path = temp_dir.path();
+    fs::write(
+        temp_dir_path.join("package.json"),
+        r#"{"scripts": {"test": "echo hello from workspace root"}}"#,
+    )
+    .unwrap();
+    fs::write(
+        temp_dir_path.join("pnpm-workspace.yaml"),
+        r#"packages: ["packages/*"]"#,
+    )
+    .unwrap();
+    let project_foo_path = temp_dir_path.join("packages/foo");
+    fs::create_dir_all(&project_foo_path).unwrap();
+    fs::write(
+        project_foo_path.join("package.json"),
+        r#"{"scripts": {"test": "echo hello from foo"}}"#,
+    )
+    .unwrap();
```

Comment:
I used to create temporary files like this a lot, so I created [build-fs-tree](https://docs.rs/build-fs-tree/0.4.1/build_fs_tree/). It is perfect for this exact purpose:

```rust
let tree = MergeableFileSystemTree::<&str, &str>::from(dir! {
    "package.json" => file!(r#"{"scripts": {"test": "echo hello from workspace root"}}"#),
    "pnpm-workspace.yaml" => file!("packages: ['packages/*']"),
    "packages/foo/package.json" => file!(r#"{"scripts": {"test": "echo hello from foo"}}"#),
});
tree.build(&temp_dir).unwrap();
```

Alternatively, if you want less magic:

```rust
let tree = dir! {
    "package.json" => file!(r#"{"scripts": {"test": "echo hello from workspace root"}}"#),
    "pnpm-workspace.yaml" => file!("packages: ['packages/*']"),
    "packages" => dir! {
        "foo" => dir! {
            "package.json" => file!(r#"{"scripts": {"test": "echo hello from foo"}}"#),
        },
    },
};
tree.build(&temp_dir).unwrap();
```